src/v/kafka/data/rpc/test/deps.h

```diff
@@ -426,7 +426,8 @@ class fake_partition_manager_proxy {
       ss::shard_id shard_id,
       const N& ntp,
       ss::noncopyable_function<
-        ss::future<result<R, cluster::errc>>(kafka::partition_proxy*)> fn) {
+        ss::future<result<R, cluster::errc>>(kafka::partition_proxy*)> fn,
+      require_leader = require_leader::yes) {
```

Comment:
I don't fully understand why we don't need to respect `require_leader` here.

Reply:
yeah I just meant in the test fixture. not sure whether it's meaningful here though.